code/modules/antagonists/eldritch_cult/eldritch_items.dm

```diff
@@ -180,3 +189,219 @@
+	if(!local_user)
+		return PROCESS_KILL
+
+	if((IS_HERETIC(local_user) || IS_HERETIC_MONSTER(local_user)) && HAS_TRAIT(src,TRAIT_NODROP))
+		REMOVE_TRAIT(src, TRAIT_NODROP, CLOTHING_TRAIT)
+
+	for(var/mob/living/carbon/human/human_in_range in spiral_range(9,local_user))
+		if(IS_HERETIC(human_in_range) || IS_HERETIC_MONSTER(human_in_range))
+			continue
+
+		SEND_SIGNAL(human_in_range,COMSIG_DRAIN_SANITY,rand(-1,-10))
+
+		if(DT_PROB(10,delta_time))
+			human_in_range.emote(pick("giggle","laugh"))
+			human_in_range.adjustStaminaLoss(10)
+
+		if(DT_PROB(25,delta_time))
+			human_in_range.Dizzy(5)
+
+		if(DT_PROB(40,delta_time))
+			human_in_range.Jitter(5)
+
+		if(DT_PROB(60,delta_time))
+			human_in_range.hallucination += 5
+
+/obj/item/melee/rune_knife
+	name = "Carving Knife"
+	desc = "Cold Steel, pure, perfect, this knife can carve the floor in many ways, but only few can evoke the dangers that lurk beneath reality."
+	icon = 'icons/obj/eldritch.dmi'
+	icon_state = "rune_carver"
+	flags_1 = CONDUCT_1
+	sharpness = SHARP_EDGED
+	w_class = WEIGHT_CLASS_SMALL
+	force = 10
+	throwforce = 5
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb_continuous = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "rends")
+	attack_verb_simple = list("attack", "slash", "stab", "slice", "tear", "lacerate", "rip", "dice", "rend")
+	var/static/list/blacklisted_turfs = typecacheof(list(/turf/closed,/turf/open/space,/turf/open/lava))
+	var/drawing = FALSE
+	var/list/current_runes = list()
+	var/max_rune_amt = 3
+	var/datum/action/innate/rune_shatter/linked_action
+
+/obj/item/melee/rune_knife/Initialize()
+	. = ..()
+	linked_action = new(src)
+
+/obj/item/melee/rune_knife/pickup(mob/user)
+	. = ..()
+	linked_action.Grant(user, src)
+
+/obj/item/melee/rune_knife/dropped(mob/user, silent)
+	. = ..()
+	linked_action.Remove(user, src)
+
+/obj/item/melee/rune_knife/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!is_type_in_typecache(target,blacklisted_turfs) && !drawing && proximity_flag)
+		carve_rune(target,user,proximity_flag,click_parameters)
+
+/obj/item/melee/rune_knife/proc/carve_rune(atom/target, mob/user, proximity_flag, click_parameters)
+	var/obj/structure/trap/eldritch/elder = locate() in range(1,target)
+	if(elder)
+		to_chat(user,"<span class='notice'>You can't draw runes that clsoe to each other!</span>")
+		return
+
+	for(var/X in current_runes)
+		var/obj/structure/trap/eldritch/eldritch = X
+		if(QDELETED(eldritch) || !eldritch)
+			current_runes -= eldritch
+
+	if(current_runes.len == max_rune_amt)
+		to_chat(user,"<span class='notice'>The blade cannot support more runes!</span>")
+		return
+
+	var/list/pick_list = list()
+	for(var/E in subtypesof(/obj/structure/trap/eldritch))
+		var/obj/structure/trap/eldritch/eldritch = E
+		pick_list[initial(eldritch.name)] = eldritch
+
+	drawing = TRUE
+
+	var/type = pick_list[input(user,"Choose the rune","Rune") in pick_list]
```

Comment:
Use `as null|anything`, and handle the `null` return, to give the user the choice to cancel the action.